### PR TITLE
Use file-name for images, eot, woff, ttf, svg and ttf

### PR DESCRIPTION
**Pull request by iLeonelPerea**
Hi @coryhouse , how can I use the filename for js and scss? I don't know how to this to their loaders...

tnks

### CODE REVIEW

**Comment by coryhouse** [comment]
May I ask why you care about the file name?

**Comment by iLeonelPerea** [comment]
because file name describes the contents of the webpage, search engines may use folder and file names to generate search results.

Why you don't care about the file name?

File: webpack.config.prod.js [hunk 1]
```diff
@@ -65,15 +65,16 @@ export default {
...
-      {test: /\.ttf(\?v=\d+\.\d+\.\d+)?$/, loader: 'url?limit=10000&mimetype=application/octet-stream'},
-      {test: /\.svg(\?v=\d+\.\d+\.\d+)?$/, loader: 'url?limit=10000&mimetype=image/svg+xml'},
-      {test: /\.(jpe?g|png|gif)$/i, loaders: ['file']},
-      {test: /\.ico$/, loader: 'file?name=[name].[ext]'},
+      {test: /\.js$/, include: path.join(__dirname, 'src'), loader: 'babel'},
+      {test: /\.eot(\?v=\d+.\d+.\d+)?$/, loader: 'url?name=[name].[ext]'},
+      {test: /\.woff(2)?(\?v=[0-9]\.[0-9]\.[0-9])?$/, loader: "url?limit=10000&mimetype=application/font-woff&name=[name].[ext]"},
+      {test: /\.ttf(\?v=\d+.\d+.\d+)?$/, loader: 'url?limit=10000&mimetype=application/octet-stream&name=[name].[ext]'},
+      {test: /\.svg(\?v=\d+.\d+.\d+)?$/, loader: 'url?limit=10000&mimetype=image/svg+xml&name=[name].[ext]'},
+      {test: /\.(jpe?g|png|gif)$/i, loader: 'file-loader?name=[name].[ext]'},
```
Comment (coryhouse) [inline webpack.config.prod.js]: Suggest changing this line to:

{test: /\.(jpe?g|png|gif)$/i, loader: 'file?name=[name].[ext]'

File: webpack.config.prod.js [hunk 2]
```diff
@@ -65,15 +65,16 @@ export default {
...
+      {test: /\.ico$/, loader: 'file-loader?name=[name].[ext]'},
```
Comment (coryhouse) [inline webpack.config.prod.js]: {test: /\.ico$/, loader: 'file?name=[name].[ext]'},

**Comment by coryhouse** [comment]
Given that the filename is randomized, I'm not sure there's much value in specifying a prefix/suffix.

That said, I'll accept the PR if you make the two tweaks I suggested (specifying the word loader isn't necessary since it's implied by convention).

**Comment by kwelch** [comment]
From a learning perspective it is good to know this is possible but not sure on benefits.

**Comment by coryhouse** [comment]
@kwelch Agreed. I can't imagine this driving any extra SEO rank. And it makes the loaders longer and harder to read. I prefer to keep things as simple as possible.

**Comment by iLeonelPerea** [comment]
@coryhouse done!! 💃

File: webpack.config.prod.js [hunk 3]
```diff
@@ -65,15 +65,16 @@ export default {
...
+      {test: /\.js$/, include: path.join(__dirname, 'src'), loader: 'babel'},
+      {test: /\.eot(\?v=\d+.\d+.\d+)?$/, loader: 'url?name=[name].[ext]'},
+      {test: /\.woff(2)?(\?v=[0-9]\.[0-9]\.[0-9])?$/, loader: "url?limit=10000&mimetype=application/font-woff&name=[name].[ext]"},
+      {test: /\.ttf(\?v=\d+.\d+.\d+)?$/, loader: 'url?limit=10000&mimetype=application/octet-stream&name=[name].[ext]'},
+      {test: /\.svg(\?v=\d+.\d+.\d+)?$/, loader: 'url?limit=10000&mimetype=image/svg+xml&name=[name].[ext]'},
+      {test: /\.(jpe?g|png|gif)$/i, loader: 'file?name=[name].[ext]'},
       {test: /\.ico$/, loader: 'file?name=[name].[ext]'},
       {
         test: /(\.css|\.scss)$/,
+        include: path.join(__dirname, 'src'),
```
Comment (coryhouse) [inline webpack.config.prod.js]: Why is adding this necessary?

**Comment by iLeonelPerea** [comment]
@coryhouse I removed the include, no really necessary.. tnks.

**Comment by coryhouse** [comment]
Thanks for the PR! 👍

**Pull request merged**